components/sc_cover/sc_cover.cpp

```diff
@@ -36,6 +37,14 @@ void SingleControlCover::setup() {
   this->open_endstop_->add_on_state_callback([this](bool state) { this->open_endstop_callback_(state); });
   this->close_endstop_->add_on_state_callback([this](bool state) { this->close_endstop_callback_(state); });
 
+  this->do_setup();
+
+  if (this->setup_delay_ > 0) {
+    this->set_timeout(this->setup_delay_, [this] { this->do_setup(); });
+  }
+}
+
+void SingleControlCover::do_setup() {
```

Comment:
I would like to follow the name pattern for private methods. Just append and underscore to the end of function name.
```suggestion
void SingleControlCover::do_setup_() {
```
Don't forget to update it in all places.

Reply:
Certainly, please hold :D